### PR TITLE
Fix broken reload screen logic

### PR DESCRIPTION
**Pull request by poelzi**
The original code fails on multi-screen setups and places the window
in between the displays. Unfortunately, there seems to be a bug in QT
which causes screen() to return stale data if the window is not moved
or clicked on after the reload while it still catches keyboard shortcuts.

Instead of replacing the window, simply safe the geometry and restore it
after reload, so the window will not move around when reloading the skin.

### CODE REVIEW

**Comment by ronso0** [comment]
please rebase on `main` to clean up the commit history.

File: src/mixxx.cpp [hunk 1]
```diff
@@ -1456,24 +1448,8 @@ void MixxxMainWindow::rebootMixxxView() {
...
-            const auto windowMarginWidth =
-                    pScreen->geometry().width() - m_pCentralWidget->width();
-            const auto windowMarginHeight =
-                    pScreen->geometry().height() - m_pCentralWidget->height();
-            newX = std::max(0, std::min(newX, windowMarginWidth));
-            newY = std::max(0, std::min(newY, windowMarginHeight));
-            move(newX, newY);
-        }
+    } else {
+        setGeometry(initGeometry);
```
Comment (daschuer) [inline src/mixxx.cpp:1452]: Please add a comment. About what will happen if the new skin minimum is bigger than the initGeometry.

**Comment by daschuer** [comment]
If this fixes the skin moving issue, it can even go to 2.2.

**Comment by poelzi** [comment]
@ronso0 i guess as long as origin/master exists, I will not learn ;)

**Comment by Holzhaus** [comment]
> @ronso0 i guess as long as origin/master exists, I will not learn ;)

    git push origin --delete master

**Comment by daschuer** [comment]
With this PR, the window is moving downwards on every call the the reload skin entry in the developers menu.

**Comment by Be-ing** [comment]
> @ronso0 i guess as long as origin/master exists, I will not learn ;)

You should branch from upstream/main anyway, not origin.

**Comment by daschuer** [comment]
Works perfectly, the window is no longer moving after skin change. 
Thank you.

**Comment by daschuer** [comment]
Be-ing changed the base branch from main to 2.3, so please: 
git rebase --onto upstream/2.3 587efb3ea6cf9aa2127b7e291b7baf9570953e3e

**Comment by Holzhaus** [comment]
I think it makes sense to document the explanation from the PR description as a comment in the source code.

**Comment by poelzi** [comment]
@Holzhaus done

**Comment by Holzhaus** [comment]
> Works perfectly, the window is no longer moving after skin change.
> Thank you.

I don't have a multi-screen setup to test this myself atm, if this works fine feel free to merge.

**Comment by uklotzde** [comment]
Thanks for the explaining the issue in a detailed comment :+1: I like this pragmatic and simple approach.

**Comment by uklotzde** [comment]
LGTM

**Pull request merged**